src/main/java/org/hl7/davinci/priorauth/ClaimEndpoint.java

```diff
@@ -287,50 +285,26 @@ private boolean processClaimItems(Bundle bundle, String id, String relatedId) {
     boolean ret = true;
     Claim claim = FhirUtils.getClaimFromRequestBundle(bundle);
     String claimStatusStr = FhirUtils.getStatusFromResource(claim);
-    for (ItemComponent item : claim.getItem()) {
-      boolean itemIsCancelled = false;
-      if (item.hasModifierExtension()) {
-        List<Extension> exts = item.getModifierExtension();
-        for (Extension ext : exts) {
-          if (ext.getUrl().equals(FhirUtils.ITEM_CANCELLED_EXTENSION_URL) && ext.hasValue()) {
-            Type type = ext.getValue();
-            itemIsCancelled = type.castToBoolean(type).booleanValue();
-          }
-        }
-      }
 
-      Disposition itemDisposition;
-      if (!itemIsCancelled) {
-        itemDisposition = PriorAuthRule.computeDisposition(bundle, item.getSequence());
-      } else
-        itemDisposition = Disposition.CANCELLED;
-
-      Map<String, Object> dataMap = new HashMap<String, Object>();
-      dataMap.put("id", id);
-      dataMap.put("sequence", item.getSequence());
-      dataMap.put("status", itemIsCancelled ? ClaimStatus.CANCELLED.getDisplay().toLowerCase() : claimStatusStr);
-      dataMap.put("outcome", FhirUtils.dispositionToReviewAction(itemDisposition).value());
-      if (relatedId != null) {
-        // This is an update
-        Map<String, Object> constraintMap = new HashMap<String, Object>();
-        constraintMap.put("id", relatedId);
-        constraintMap.put("sequence", item.getSequence());
-
-        // Update if item exists
-        if (App.getDB().readStatus(Table.CLAIM_ITEM, constraintMap) != null) {
-          if (!App.getDB().update(Table.CLAIM_ITEM, constraintMap, dataMap)) {
-            logger.warning(
-                "ClaimEndpoint::processClaimItems:unable to update claim item:" + id + "/" + item.getSequence());
-            ret = false;
-          }
-          continue;
-        }
-      }
+    // Start all of the threads
+    Map<Integer, ProcessClaimItemTask> threads = new HashMap<Integer, ProcessClaimItemTask>();
+    for (ItemComponent item : claim.getItem()) {
+      ProcessClaimItemTask itemTask = new ProcessClaimItemTask(bundle, item, id, relatedId, claimStatusStr);
+      threads.put(item.getSequence(), itemTask);
+      itemTask.start();
+    }
 
-      // Add new item to the database
-      if (!App.getDB().write(Table.CLAIM_ITEM, dataMap)) {
-        logger.warning("ClaimEndpoint::processClaimItems:unable to write claim item:" + id + "/" + item.getSequence());
+    // Block until all of the threads are done
+    for (ProcessClaimItemTask itemTask : threads.values()) {
+      try {
+        itemTask.getThread().join();
+        if (itemTask.getStatus() != 0)
+          ret = false;
+        logger.fine("ClaimEndpoint::processClaimItems:finsihed processing " + itemTask.getItemName() + ":"
+            + itemTask.getStatus());
+      } catch (InterruptedException e) {
```

Comment:
Are there any other kinds of exceptions we should catch?

Reply:
This is the only one I found online we should be catching